src/h5web/visualizations/heatmap/utils.ts

```diff
@@ -1,35 +1,48 @@
 import React, { useCallback, useRef } from 'react';
 import { extent } from 'd3-array';
 import { rgb } from 'd3-color';
-import { scaleSequential } from 'd3-scale';
-import { interpolateMagma } from 'd3-scale-chromatic';
+import { scaleSequential, ScaleSequential, scaleLinear } from 'd3-scale';
 import { Vector3 } from 'three';
 import { ReactThreeFiber, PointerEvent, useThree } from 'react-three-fiber';
 import { clamp } from 'lodash-es';
 
 const ZOOM_FACTOR = 0.95;
 
-export function computeTextureData(matrix: number[][]): Uint8Array | undefined {
-  const values = matrix.flat();
-  const [min, max] = extent(values);
-
-  if (min === undefined || max === undefined) {
+export function computeTextureData(
+  values: number[],
+  scale?: ScaleSequential<string>
+): Uint8Array | undefined {
```

Comment:
Might be easier to require `scale` and call this function like so: `const textureData = scale && computeTextureData(...)`